### PR TITLE
docs: Add more details for the Cluster Mesh key rotation

### PR DESCRIPTION
**Pull request by margamanterola**
Rather than having this information hidden in a parenthesis in the troubleshooting section, let's have a full paragraph in the key rotation section that explains what needs to be done when using Cluster Mesh.

### CODE REVIEW

File: Documentation/security/network/encryption-ipsec.rst [hunk 1]
```diff
@@ -182,6 +182,11 @@ included and should be monotonically increasing every re-key with a rollover
 from 15 to 1. The Cilium agent will default to ``KEYID`` of zero if its not
 specified in the secret.
 
+If you are using Cluster Mesh, the key rotation procedure needs to be applied
+to all clusters in the mesh. You might need to adjust the transition time to
```
Comment (jschwinger233) [inline Documentation/security/network/encryption-ipsec.rst]: How about making it more specific, `adjust` -> `increase`?

**Review by pchaigno — APPROVED** [review]
Thank you!

File: Documentation/security/network/encryption-ipsec.rst (hunk 1 above)
Comment (margamanterola) [reply Documentation/security/network/encryption-ipsec.rst]: Done, thanks.

**Review by jschwinger233 — APPROVED** [review]

File: Documentation/security/network/encryption-ipsec.rst [hunk 2]
```diff
@@ -182,6 +182,11 @@ included and should be monotonically increasing every re-key with a rollover
 from 15 to 1. The Cilium agent will default to ``KEYID`` of zero if its not
 specified in the secret.
 
+If you are using Cluster Mesh, the key rotation procedure needs to be applied
+to all clusters in the mesh. You might need to increase the transition time to
+allow for the new keys to get deployed and applied across all clusters,
+you can do that with the agent flag ``ipsec-key-rotation-duration``.
```
Comment (zacharysarah) [inline Documentation/security/network/encryption-ipsec.rst]: Minor edits for clarity
```suggestion
If you are using Cluster Mesh, you must apply the key rotation procedure
to all clusters in the mesh. You might need to increase the transition time to
allow for new keys to be deployed and applied across all clusters,
which you can do with the agent flag ``ipsec-key-rotation-duration``.
```

**Review by zacharysarah — CHANGES_REQUESTED** [review]
@margamanterola Valuable PR! ✨ Some small edits for clarity, otherwise LGTM

File: Documentation/security/network/encryption-ipsec.rst (hunk 2 above)
Comment (margamanterola) [reply Documentation/security/network/encryption-ipsec.rst]: Applied, thanks.

**Review by zacharysarah — APPROVED** [review]
LGTM

**Comment by margamanterola** [comment]
This is a documentation-only check, so no need to waste energy running conformance tests. Setting ready-to-merge.

**Pull request merged**